File: solutions/javascript/0338-counting-bits.js
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
    let res = new Array(n + 1).fill(0);
    for(let i = 1; i <= n; i++)
        res[i] = res[i >> 1] + (i & 1);
    return res;
};

File: solutions/csharp/0338-counting-bits.cs
public class Solution {
    public int[] CountBits(int n) {
        int[] res = new int[n + 1];
        for(int i = 1; i <= n; i++)
            res[i] = res[i >> 1] + (i & 1);
        return res;
    }
}
